torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -1514,4 +1517,170 @@ XLAGraphExecutor::SyncTensorsGraphInternal(
   }
 }
 
+// std::shared_ptr<XLAGraphExecutor::Async>
+// XLAGraphExecutor::CompilationResult
+// runtime::ComputationClient::ComputationPtr
+// XLAGraphExecutor::saveComputation*
+// std::vector<runtime::ComputationClient::ComputationPtr> 
+XLAGraphExecutor::CachedComputation XLAGraphExecutor::GetXLAComputation(std::vector<XLATensorPtr>* tensors,
+    absl::Span<const std::string> devices, bool warm_up_cache_only = false) {
+  tsl::profiler::TraceMe activity("GetXLAComputation",
+                                  tsl::profiler::TraceMeLevel::kInfo);
+  SyncTensorsConfig config;
+  config.force_ltc_data = false;
+  SyncTensorCollection coll = CollectSyncTensors(*tensors, config);
+  if (coll.indices.empty()) {
+    TensorCollectionBarrier(&coll);
+    return nullptr;
+  }
+  DebugUtil::SaveTensorsGraphInfo("ScheduleSyncTensorsGraph", *tensors,
+                                  &coll.indices);
+  std::vector<torch::lazy::Value> ir_values;
+  std::vector<torch::lazy::BackendDataPtr> tensor_data_vec;
+  ExtractIRAndPrepareXlaData_(tensors, coll.config, coll.indices, ir_values,
+                              tensor_data_vec);
+  PostOrderData po_data = RunPostOrder(ir_values, &coll);
+  coll.hash = torch::lazy::HashCombine(
+      coll.hash, torch::lazy::Hash(po_data.parameter_sequence));
+  if (GetAliasWithBufferDonorConfig()) {
+    std::vector<size_t> buffer_donor_index =
+        GetBufferDonorIndexFromUserConfig(po_data.parameters_data);
+    if (buffer_donor_index.size() > 0) {
+      // Do not include hash on a empty vector.
+      coll.hash = torch::lazy::HashCombine(
+          coll.hash, torch::lazy::Hash(buffer_donor_index));
+    }
+  }
+  {
+    // Auto-sharding configs
+    coll.hash = torch::lazy::HashCombine(
+        coll.hash, torch::lazy::MHash(ShardingUtil::GetAutoSharding()));
+    coll.hash = torch::lazy::HashCombine(
+        coll.hash,
+        torch::lazy::StringHash(
+            runtime::sys_util::GetEnvString("XLA_AUTO_SPMD_MESH", "").c_str()));
+  }
+
+  DebugUtil::SaveGraphHash(coll.hash);
+  TF_VLOG(4) << "Parameter sequence graph hash "
+             << torch::lazy::HashToString(coll.hash);
+
+  std::pair<bool, std::shared_ptr<XLAGraphExecutor::Async>> cache_res =
+      TryRunCachedSync(tensors, &coll, &po_data, tensor_data_vec,
+                       warm_up_cache_only);
+  if (cache_res.first) {
+    // we have a cache hit, execution has been scheduled by TryRunCachedSync.
+    return cache_res.second;
+  }
+
+  // CompilationResult compile_result =
+  //     Compile(*tensors, devices, coll, &po_data, ir_values);
+
+  // runtime::ComputationClient::ComputationPtr 
+  // saveComputation* compile_result = std::move(
+  //   Compile(*tensors, devices, coll, &po_data, ir_values).computation)
+  XLAGraphExecutor::saveComputation* compile_result = Compile(*tensors, devices, coll, &po_data, ir_values).computation
+  return compile_result
+
```

Comment:
Hi, @miladm, this is the script code implementation for post-order tracing compile function